File: 151-reverse-words-in-a-string/reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        stringstream ss(s);
        string result;
        string word;
        while(ss>>word){
            result = word +" " + result;

        }
        if(!result.empty()){
            result.pop_back();
        }
        return result;
        
    }
};

File: 151-reverse-words-in-a-string/reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
          String str[] = s.trim().split("\\s+");
        String words ="";
        for(int i=str.length-1;i>=0;i--){
            words += str[i] + " ";
        }
        return words.trim();

    }
}